acertmgr/tools.py

```diff
@@ -257,15 +258,31 @@ def get_cert_valid_until(cert):
 
 
 # @brief convert certificate to PEM format
-# @param cert certificate object in pyopenssl format
+# @param cert certificate object or a list thereof
 # @return the certificate in PEM format
 def convert_cert_to_pem_str(cert):
-    return cert.public_bytes(serialization.Encoding.PEM).decode('utf8')
+    if isinstance(cert, list):
+        result = list()
+        for data in cert:
+            result.append(data.public_bytes(serialization.Encoding.PEM).decode('utf8'))
+        return '\n'.join(result)
+    else:
+        return cert.public_bytes(serialization.Encoding.PEM).decode('utf8')
```

Comment:
Do both of these cases occur in practice?

Reply:
They will, once a CA chain is passed to convert_pem_str_to_cert which will be passed through in the rest of the code and land here for writing it back out to a file.